### PR TITLE
Bump node to 1.35.3 and switch to preview

### PR DESCRIPTION
**Pull request by t4ccer**
Not sure if you want it, if not please do not remove my branch

### CODE REVIEW

File: example/BPI/Testnet/Setup.hs [hunk 1]
```diff
@@ -70,7 +73,11 @@ runSetup workDir = do
   setLocaleEncoding utf8
   createRequiredDirs
   sockPath <- getEnv "CARDANO_NODE_SOCKET_PATH"
-  let nodeInfo = BPI.NodeInfo (Testnet $ NetworkMagic 2) sockPath
+  let nodeInfo =
+        LocalNodeConnectInfo
+          (CardanoModeParams $ EpochSlots (60 * 60 * 24))
+          (Testnet $ NetworkMagic 2)
+          sockPath
```
Comment (TotallyNotChase) [inline example/BPI/Testnet/Setup.hs:80]: Are these changes necessary because of upstream BPI changes? I kind of missed them

File: example/frontend/src/index.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ function getWalletObj(x: WalletObjContainer) {
 const blockfrostInst = new Blockfrost(blockfrostUrl, blockfrostProjId);
 
 // Initialise the lucid instance.
-const lucid = await Lucid.new(blockfrostInst, "Testnet");
+const lucid = await Lucid.new(blockfrostInst, "Preview");
```
Comment (TotallyNotChase) [inline example/frontend/src/index.ts:21]: Need to update the lucid dep for this to work

File: flake.nix [hunk 3]
```diff
@@ -5,9 +5,10 @@
     nixpkgs.follows = "bot-plutus-interface/nixpkgs";
     haskell-nix.follows = "bot-plutus-interface/haskell-nix";
 
-    bot-plutus-interface.url = "github:mlabs-haskell/bot-plutus-interface?ref=gergely/vasil";
-
     cardano-node.url = "github:input-output-hk/cardano-node?ref=1.35.3";
+
+    bot-plutus-interface.url = "github:mlabs-haskell/bot-plutus-interface";
+    bot-plutus-interface.inputs.cardano-node.follows = "cardano-node";
```
Comment (TotallyNotChase) [inline flake.nix:11]: I guess the `cardano-node` input in the flake here should just follow `bot-plutus-interface.inputs.cardano-node` - I reckon  it'd be the same in this case? Can we trust BPI to always follow the right node input?

File: src/Plutus/Contract/PartialTx.hs [hunk 4]
```diff
@@ -232,7 +232,7 @@ data PartialTxIn = PartialTxIn
 -- | Detailed information about the type of a 'TxIn', and related info.
 data PartialTxInDetailed
   = ScriptTxIn
-      { ptxInDet'version :: LedgerPlutusVersion
+      { ptxInDet'version :: Language
```
Comment (TotallyNotChase) [inline src/Plutus/Contract/PartialTx.hs:235]: Is this necessary?

**Review by TotallyNotChase — CHANGES_REQUESTED** [review]

File: example/BPI/Testnet/Setup.hs (hunk 1 above)
Comment (t4ccer) [reply example/BPI/Testnet/Setup.hs:80]: Yes, `BPI.queryProtocolParams` is no longer a thing and you need this to run query effect

File: flake.nix (hunk 3 above)
Comment (t4ccer) [reply flake.nix:11]: rn it's the same, but I don't know if we can always trust BPI

File: src/Plutus/Contract/PartialTx.hs (hunk 4 above)
Comment (t4ccer) [reply src/Plutus/Contract/PartialTx.hs:235]: Yes, `LedgerPlutusVersion` defined in `Plutus.ApiCommon` for some reason doesn't have `Show` instance that we use. `Language` has constructors with the same names as `LedgerPlutusVersion` but with `Show`

File: example/frontend/src/index.ts (hunk 2 above)
Comment (t4ccer) [reply example/frontend/src/index.ts:21]: Done!

**Review by TotallyNotChase — APPROVED** [review]
Looks good to me. Currently have to wait for nix to allow me to test the changes here - will merge afterwards.
